### ISSUE TITLE
Selection meta-data returns wrong layer ids

### ISSUE THREAD

**Issue by umarniz** [adobe-photoshop/generator-core#296]
I am parsing metadata to know which layer is currently selected in photoshop and then saving settings generator settings per layer.

I just noticed that the onimagechange returns selected layer ids array starting from 0 but when i pull full document info i get layer starting from 1.

When I use setLayerSettingsForPlugin and pass it the layer id 0, it does not produce any errors and instead goes on and acts like setDocumentSettingsForPlugin.

I see the second error occurs getGeneratorSettings extend script has the following condition:

if (params.layerId) {
    theRef.putIdentifier(classLayer, params.layerId);
}

Which would fail when layer id is 0 and would instead treat is a DocumentSetting.

I have temporarily fixed the error by adding 1 to the layer id reported in the meta-data.

**Comment by umarniz** [comment]
An update on this, the selection layers actually return layer index instead of layer ID. My test PSD was ambiguous, hence I didn't notice the difference.

It would make more sense to send layer IDs as that is what is used to reference layers through out the generator API but at-least its not a bug :)

**Comment by joelrbrandt** [comment]
I agree that it would be nice to return layer IDs instead of (or in addition to) indexes. I suspect the use of indexes is for efficiency (is that correct, @timothynoel )?

At any rate, since this is a feature request and not a bug, I'm going to a.) mark it as such, and b.) change the title of the issue so that others aren't confused.

**Comment by timothynoel** [comment]
Possible, it would just be a matter of sending more stuff to send back (we can get ids from indeces in constant time.) We'd need to send both id/indeces I think, and put the id's into another array to preserve compatibility. You still need the indeces tell you where the layers went, or you'd have to send back the full list.   I'll keep it in mind, but it will have to wait for the next version.

**Comment by joelrbrandt** [comment]
@timothynoel In a perfect world, we could do this entirely on the JS side. However, we haven't yet moved to a world where generator-core keeps a document model around (that code is still in generator-assets). If that code were moved to core, it would be easy to do this rewriting in JS.

As you know, though, we've all got a million things to do, and moving to a stateful model in generator-core is pretty low on that to do list. :smile:
